src/Microsoft.Data.Analysis/DataFrameColumnCollection.cs

```diff
@@ -141,5 +141,73 @@ protected override void ClearItems()
             _columnNames.Clear();
             _columnNameToIndexDictionary.Clear();
         }
+
+        /// <summary>
+        /// Searches for a <see cref="PrimitiveDataFrameColumn{T}"/> with the specified <paramref name="name"/> and attempts to return it as <see cref="PrimitiveDataFrameColumn{T}"/>. If <typeparamref name="T"/> does not match <see cref="DataFrameColumn.DataType"/>, an exception is thrown.
+        /// </summary>
+        /// <typeparam name="T">The type to cast the column to</typeparam>
+        /// <param name="name">The name of the column</param>
+        /// <returns><see cref="PrimitiveDataFrameColumn{T}"/> with <paramref name="name"/></returns>
+        /// <exception cref="ArgumentException">Throws this exception if types don't match</exception>
+        public PrimitiveDataFrameColumn<T> GetPrimitiveColumn<T>(string name)
+            where T : unmanaged
+        {
+            int columnIndex = IndexOf(name);
```

Comment:
We should have an indexer by column name, so we don't need to duplicate this logic every time:

```C#
int columnIndex = IndexOf(name);
            if (columnIndex == -1)
            {
                throw new ArgumentException(Strings.InvalidColumnName, nameof(name));
            }
            DataFrameColumn column = this[columnIndex];
```